### PR TITLE
KOGITO-4477 - Operator sees the Infinispan as not ready after it is r…

### PR DESCRIPTION
**Pull request by MarianMacik**
…estarted

Many thanks for submitting your Pull Request :heart:! 

[KOGITO-4477](https://issues.redhat.com/browse/KOGITO-4477)

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: controllers/infinispan.go [hunk 1]
```diff
@@ -390,9 +391,9 @@ func (i *infinispanInfraReconciler) Reconcile() (requeue bool, resultErr error)
 			return true, nil
 		}
 	}
-	infinispanCondition := getLatestInfinispanCondition(infinispanInstance)
+	infinispanCondition := getInfinispanWellFormedCondition(infinispanInstance)
 	if infinispanCondition == nil || infinispanCondition.Status != string(v1.ConditionTrue) {
-		return false, errorForResourceNotReadyError(fmt.Errorf("infinispan instance %s not ready. Waiting for Condition.Type == True", infinispanInstance.Name))
+		return false, errorForResourceNotReadyError(fmt.Errorf("infinispan instance %s not ready. Waiting for Condition with Condition.Type == wellFormed to have Condition.Status == True", infinispanInstance.Name))
```
Comment (sutaakar) [inline controllers/infinispan.go:396]:
```suggestion
		return false, errorForResourceNotReadyError(fmt.Errorf("infinispan instance %s not ready. Waiting for Condition with Condition.Type == %s to have Condition.Status == True", infinispanInstance.Name, infinispanConditionWellFormed))
```

Comment (sutaakar) [reply controllers/infinispan.go:396]: Just a nitpick

**Review by sutaakar — APPROVED** [review]
Looks good.
I suppose you tested it properly against Infinispan.

**Comment by MarianMacik** [comment]
I tested it locally in a custom project and it is able to correctly reconcile the Infinispan KogitoInfra. BDD tests failed because of misalignment of the https://github.com/kiegroup/kogito-runtimes/pull/1069 between kogito-runtimes and kogito-examples. Kogito-runtimes is not yet in Nexus.

**Comment by MarianMacik** [comment]
@ricardozanini @spolti Can you please have a look as well so we can merge? I don't want to merge this without blessing of at least one of you :)

**Comment by ricardozanini** [comment]
Great catch man! Many thanks. The missing `latestConditionData` is really cumbersome. I believe we could open an issue for them to add this field.

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
Hold until #740 
I believe we can wait one more day since the refactoring is probably about to be merged.

**Review by spolti — APPROVED** [review]

**Comment by spolti** [comment]
@MarianMacik nice one :)
can you please rebase this pr as well?

**Comment by MarianMacik** [comment]
Rebased.
@vaibhavjainwiz Just one question. Why was the `Test_Reconcile_Infinispan` moved to a separate file `infinispan_reconciler_test.go`? Because in the old file [kogitoinfra_controller_test.go](https://github.com/kiegroup/kogito-cloud-operator/blob/master/controllers/kogitoinfra_controller_test.go) it was testing also the calling `(r *KogitoInfraReconciler) Reconcile(req ctrl.Request)` method, so the coverage was better a little bit.

**Comment by ricardozanini** [comment]
@MarianMacik, @vaibhavjainwiz's answer to your question block us from merging this PR?

**Comment by vaibhavjainwiz** [comment]
> @MarianMacik, @vaibhavjainwiz's answer to your question blocks us from merging this PR?

I already had discussion on this point with @MarianMacik today's afternoon on Zulip. I am attaching our discussion:
```
Marian Macik: Hi Vaibhav, can you please answer my question on this PR: https://github.com/kiegroup/kogito-cloud-operator/pull/767#issuecomment-783660267
Thank you!

Vaibhav Jain: Sorry @Marian Macik , I miss your comment on PR

Vaibhav Jain: I keep Test_Reconcile_KafkaResource testcase in kogitoinfra_controller_test.go to test end to end flow of Infra controller. Apart for that test I moved all testcase to there respective reconciler file like in infinispan_reconciler_test.go, kafka_reconciler_test.go etc so that developer could relates its test scenerios

Marian Macik: ok, makes sense now, thanks!
```

**Comment by MarianMacik** [comment]
@ricardozanini No, it wasn't blocking. Merging it now. Thank you guys!

**Pull request merged**